File: GameLibrary/src/cc/lib/reflector/IDirty.java
package cc.lib.reflector;

import java.io.IOException;

/**
 * Created by Chris Caron on 11/30/23.
 */
public interface IDirty {
    boolean isDirty();

    void markClean();

    void serializeDirty(RPrintWriter out) throws IOException;
}

File: GameLibrary/src/cc/lib/kreflector/IDirty.kt
package cc.lib.kreflector

import java.io.IOException

/**
 * Created by Chris Caron on 11/30/23.
 */
interface IDirty {
	fun isDirty(): Boolean
	fun markClean()

	@Throws(IOException::class)
	fun serializeDirty(out: RPrintWriter)
}